### ISSUE TITLE
New Azkaban DSL

### ISSUE THREAD

**Issue by davidzchen** [azkaban/azkaban#230]
The current way of describing flows using Java Properties-based .job files is not ideal. As flows become more complex, the number of job files increases, and it becomes increasingly difficult to manage and debug issues, such as unintentional circular dependencies. The DSL should also be easily extensible.

A new DSL for describing Azkaban flows has been on the Azkaban wishlist for a long time and will make Azkaban even more easy and productive for users. Creating a new DSL will also allow us to address issues that have been brought up, such as arithmetic expressions (#224) and being able to set a custom name for a flow (#41).

The code for processing the new DSL should also be reusable enough so that it can be integrated into an Azkaban Gradle plugin to validate Azkaban projects locally without having to deploy them.

**Comment by pranayhasan** [comment]
github.com/linkedin/linkedin-gradle-plugin-for-apache-hadoop

**Comment by davidzchen** [comment]
@pranayhasan Yes, I am aware that this plugin exists, but IMO workflow code should not be implemented by the build system.
